src/main/java/io/cdap/plugin/marketo/common/api/Helpers.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.marketo.common.api;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import io.cdap.plugin.marketo.common.api.entities.DateRange;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Various helper methods.
+ */
+public class Helpers {
+  public static String streamToString(InputStream inputStream) {
+    try {
+      return IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public static <T> T streamToObject(InputStream inputStream, Class<T> cls) {
+    return Marketo.GSON.fromJson(new InputStreamReader(inputStream), cls);
+  }
+
+  public static RuntimeException failForUri(String method, URI uri, Exception ex) {
+    String message = ex.getMessage();
+    if (Strings.isNullOrEmpty(message)) {
+      if (ex.getCause() != null) {
+        message = ex.getCause().getMessage();
+        if (Strings.isNullOrEmpty(message)) {
+          message = "failed to make request";
+        }
+      }
+    }
+
+    URIBuilder uriBuilder = new URIBuilder(uri);
+    List<NameValuePair> queryParameters = uriBuilder.getQueryParams();
+    queryParameters.removeIf(queryParameter -> queryParameter.getName().equals("access_token"));
+    uriBuilder.setParameters(queryParameters);
+    try {
+      String uriString = uriBuilder.build().toString();
+      return new RuntimeException(String.format("Failed %s %s - %s", method, uriString, message));
```

Comment:
This will be the best message possible - "Failed GET http://example.com - Some exception message". This kind of failures can't be enriched or processed somehow in context of Marketo api.